projects/packages/my-jetpack/_inc/data/utils/get-guessed-site-lifecycle-status.ts

```diff
@@ -0,0 +1,35 @@
+const getGuessedSiteLifecycleStatus = () => {
+	const lifecycleStats = window?.myJetpackInitialState?.lifecycleStats;
```

Comment:
I'm thinking we could create a hook `useWindowStateQuery()` that also guards against any errors (e.g., when the state is not found).

Or do you think it's an overkill?

Reply:
Also, if `lifecycle === undefined`, maybe we can return something instead of continuing the function?

E.g. `return 'unknown'`

I'm getting more thoughts as I'm reviewing the code I created ha 😅 In my original implementation I think I assumed `lifecycleStats` are defined.

Reply:
> I'm thinking we could create a hook useWindowStateQuery() that also guards against any errors (e.g., when the state is not found).

I like this idea, but since we use the window data in other places it might be best to handle this in a cleanup task where we can replace all instances of using global data with that hook. Does that sound like a good idea?